csrc/scheduler/matmul_utils.cpp

```diff
@@ -750,14 +760,21 @@ std::unique_ptr<MatmulParams> getMatmulHeuristics(
         problem_shape[(size_t)MatmulDimRole::Batch],
         inner_dims,
         tensor_roles);
+    // TODO: more sophisticated handling of multiple matmuls when using plugin
+    mparams->tile_sizes.cta_tile.m /= patterns.size();
```

Comment:
Ideally the heuristic would understand that we might have multiple matmuls being computed in a single main loop, and that the operands for that main loop can be loaded simultaneously. For example if the A operand is used in two matmuls then we will have 3 operands loaded instead of 3, meaning we should make the CTA tile at most 2/3 as large as it would be for a single matmul. This is more conservative for now to ensure we don't run out of smem.